src/cargo/core/features.rs

```diff
@@ -371,16 +373,33 @@ impl CliUnstable {
             );
         }
         for flag in flags {
-            self.add(flag)?;
+            let mut parts = flag.splitn(2, '=');
+            let k = parts.next().unwrap();
+            let v = parts.next();
+            self.add(k, v)?;
         }
         Ok(())
     }
 
-    fn add(&mut self, flag: &str) -> CargoResult<()> {
-        let mut parts = flag.splitn(2, '=');
-        let k = parts.next().unwrap();
-        let v = parts.next();
+    /// Read unstable options from a hashmap, updating in-place.
+    /// Intended for consuming unstable settings from config files
+    pub fn update_with_table(&mut self, flags: &HashMap<String, String>) -> CargoResult<()> {
+        if !flags.is_empty() && !nightly_features_allowed() {
```

Comment:
We discussed this a bit at the Cargo meeting, and I think actually the best thing to do here is to just ignore the table on the non-nightly channel. That should make it much more convenient for users testing out nightly features because all you need to do is set it in `$HOME/.cargo/config.toml` and forget about it.

Reply:
Sounds good to me — I'll add tests to assert both sides of that behavior.